ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -118,6 +119,9 @@ public struct SignalProducer<T, E: ErrorType> {
 			for event in events {
 				observer.put(event)
 			}
+			if let termEvent = terminationEvent {
```

Comment:
You can use `let terminationEvent` here, to shadow the variable with the unwrapped version.